### PR TITLE
New slots api

### PR DESCRIPTION
**Pull request by ArtoLord**

### CODE REVIEW

File: lzy/slots-api/src/main/proto/ai/lzy/v1/slots/slots-api-v2.proto [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+option java_outer_classname = "LSA";
+option java_package = "ai.lzy.v1.slots.v2";
+
+service LzySlotsApi {
+
+  // Connect slot to remote peer (call from channel manager)
+  // This operation will continue until all the data has been sent/received.
+  // If error occurs, operation must fail with some reasonable description
+  // Cancel must be implemented.
+  rpc ConnectSlot (ConnectSlotRequest) returns (ai.lzy.v1.longrunning.Operation) {
```
Comment (ArtoLord) [inline lzy/slots-api/src/main/proto/ai/lzy/v1/slots/slots-api-v2.proto:19]: ConnectPeer

File: lzy/slots-api/src/main/proto/ai/lzy/v1/slots/slots-api-v2.proto [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+message ConnectSlotRequest {
+  string peer_id = 1;
+
+  channel.v2.PeerDescription target = 2;
+}
+
+message ConnectSlotResponse {
+}
+
+message ConnectSlotMetadata {
```
Comment (ArtoLord) [inline lzy/slots-api/src/main/proto/ai/lzy/v1/slots/slots-api-v2.proto:41]: Add more info here
like peer_id, role, ...

File: lzy/slots-api/src/main/proto/ai/lzy/v1/slots/slots-api-v2.proto (hunk 1 above)
Comment (ArtoLord) [reply lzy/slots-api/src/main/proto/ai/lzy/v1/slots/slots-api-v2.proto:19]: Maybe rename to smth else)

File: lzy/channel-manager-api/src/main/proto/ai/lzy/v1/channel/v2/channel-manager-private.proto [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import "ai/lzy/v1/common/channel.proto";
+import "ai/lzy/v1/long-running/operation.proto";
+import "ai/lzy/v1/long-running/option.proto";
+import "ai/lzy/v1/common/data-scheme.proto";
+
+option java_outer_classname = "LCMPS";
+option java_package = "ai.lzy.v1.channel.v2";
+
+service LzyChannelManagerPrivate {
+  rpc Create (CreateRequest) returns (CreateResponse);
```
Comment (imakunin) [inline lzy/channel-manager-api/src/main/proto/ai/lzy/v1/channel/v2/channel-manager-private.proto:14]: `GetOrCreate` ?

File: lzy/channel-manager-api/src/main/proto/ai/lzy/v1/channel/v2/channel-manager-private.proto [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+message CreateRequest {
+  string execution_id                                     = 1;
+  string workflow_name                                    = 2;
+  string user_id                                          = 3;
+
+  /* optional */ai.lzy.v1.common.DataScheme scheme        = 4;  // If not set, using plain DataScheme
+
+  // Initial storage peer
+  oneof storage_peer {
```
Comment (imakunin) [inline lzy/channel-manager-api/src/main/proto/ai/lzy/v1/channel/v2/channel-manager-private.proto:45]: s/storage_peer/initial_storage_peer/

File: lzy/channel-manager-api/src/main/proto/ai/lzy/v1/channel/v2/channel-manager-private.proto [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+message CreateResponse {
+  string channel_id = 1;
+}
+
+// ============== Destroy ==================
+
+message DestroyRequest {
+  string channel_id = 1;
+  string cause      = 2;
```
Comment (imakunin) [inline lzy/channel-manager-api/src/main/proto/ai/lzy/v1/channel/v2/channel-manager-private.proto:59]: s/cause/reason

empty string stands for no error, doesn't it?

Comment (ArtoLord) [reply lzy/channel-manager-api/src/main/proto/ai/lzy/v1/channel/v2/channel-manager-private.proto:59]: `reason` is just a human-readable string, which is used for logging purposes. To control behavior, the `force` flag can be used.

File: lzy/slots-api/src/main/proto/ai/lzy/v1/slots/slots-api-v2.proto [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  // Read data from slot (call from peer)
+  rpc Read (ReadDataRequest) returns (stream ReadDataChunk);
+}
+
+// ============ Connect ==============
+
+message ConnectPeerRequest {
+  string peer_id = 1;
```
[the remark this answers is not in the document]
Comment (ArtoLord) [reply lzy/slots-api/src/main/proto/ai/lzy/v1/slots/slots-api-v2.proto:33]: This is id of slot on this worker. Slots api will find slot instance by id and connect it to target.

**Review by imakunin — APPROVED** [review]

**Pull request merged**